### ISSUE TITLE
Chronos removes first character from OOO reason percentage

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#24304]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open chat with Chronos (internal tool)
2. Write "ooo 2023-08-02 75% For some reason" where the date is a current or future date.

## Expected Result:
An OOO event is added on that date with the text after the date as the title of the event

## Actual Result:
The first character, "7" is removed from the percentage in the title

## Workaround:
unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.51-2
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
![image - 2023-08-08T225852 145](https://github.com/Expensify/App/assets/43995119/1479c75c-7af2-4d85-b7bf-88362d1131ff)
![Screen Shot 2023-08-08 at 10 57 57 PM](https://github.com/Expensify/App/assets/43995119/b618a89a-f2bd-4c9c-8232-b2b802f66fa6)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @neil-marcellini 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690990218731259

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by greg-schroeder** [comment]
Hmm. I can't reproduce this

**Comment by greg-schroeder** [comment]
Nevermind, I was able to reproduce it after all! Not sure if this should be External or Internal though as I'm not sure Chronos' role in the general product vs. internal use

**Comment by greg-schroeder** [comment]
I chatted with @thienlnam and I think this is internal

**Comment by greg-schroeder** [comment]
Thoughts on how to fix this one @techievivek?

**Comment by techievivek** [comment]
Sorry for missing this, taking a look today.

**Comment by techievivek** [comment]
I see. If I use `working`then this seems to work just fine; otherwise, it strips the first character from the percentage.

<img width="1184" alt="Screenshot 2023-08-15 at 11 57 37 AM" src="https://github.com/Expensify/App/assets/35863227/97e94fe3-8e91-44d2-aae9-c80460a6e3df">

**Comment by greg-schroeder** [comment]
Yup that was my experience as well

**Comment by techievivek** [comment]
Ok, I dug up this a bit and figured out that when we have the percentage right after the date, then that gets seen as `hour`, and that's why the first digit gets cut.

**Case 1**: With some content between the date and percentage part
![Screenshot 2023-08-15 at 12 16 50 PM](https://github.com/Expensify/App/assets/35863227/f6ee4c51-ec46-426f-90a5-89e533f53887)

**Case 2**: When we have the percentage right after the date part
![Screenshot 2023-08-15 at 12 11 20 PM](https://github.com/Expensify/App/assets/35863227/27183569-6b04-49ea-8d86-e2ff1500169e)


The code isn't touched in a while so maybe this is intended? I am going to cc @tgolen to confirm if this is a bug or a feature.

**Comment by techievivek** [comment]
Code for this part is here: https://github.com/Expensify/Web-Expensify/blob/fb5cfc6c1740578084214db98bd14230e5cdc977/lib/Chronos/Plugins/OutOfOffice.php#L180

**Comment by tgolen** [comment]
Oh, funny, we were just talking about this over in https://github.com/Expensify/Expensify/issues/308284#issuecomment-1679384544 which looks like it's a duplicate. 

Vivek, you are 100% correct in the root cause.

I wouldn't say it was necessarily intentional, but it's a complex regex that has pretty low value in trying to get absolutely perfect every time. I just tell people to not put the X% at the beginning of the reason portion, put it at the end. It's a quick solution. However, if someone really feels strongly about spending time to update the regex, here it is in all it's glory:

https://github.com/Expensify/Web-Expensify/blob/47f5d938ded5eae62d88e0c83cda89b713c47987/lib/Chronos/Plugins/OutOfOffice.php#L181-L189

**Comment by greg-schroeder** [comment]
Thanks for that context, Tim! I think it feels pretty low value personally and our time might be better spent elsewhere as there's a very viable workaround. It also doesn't effect the main use case where it would cause a problem (which is when you want to set it to 0%)

**Comment by techievivek** [comment]
I agree that fixing the regex might take some time(even then we can't guarantee it will be 100% working for all the cases), so since it's a low-value concern with no side effects, I suggest we do nothing here.
